### ISSUE TITLE
[Feature request]: Allow purge of system installation without password prompt

### ISSUE THREAD

**Issue by mcrha** [flatpak/flatpak#5251]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

This is related to the gnome-software:
https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2027

There had been added automatic purge of unused refs into the gnome-software, but it works only for the user installation, because the removal requires password. It would be good to be able to do the purge without asking for the password, the same as operations like app/runtime updates.

The way the gnome-software implements the purge is as:

* get the list of unused refs with `flatpak_installation_list_unused_refs()`
* create a `no_interaction` and `no_pull` transaction
* add listed unused refs as `uninstall` to the transaction
* run the transaction

I can understand the distinguish between ordinary uninstall and purge of the unused refs can be tricky, but maybe if the flatpak checks whether the transaction contains only unused refs, then it can decide whether the password is needed or not.

**Comment by JakobDev** [comment]
I don't know if it is a good idea to allow changes on the System without a prompt

**Comment by JakobDev** [comment]
The problem is, that you are still running a program as root without asking for permission. If this function has some security issue, this could be a problem.

**Comment by mcrha** [comment]
Right, it's why I was not sure how to do it to avoid the (security) problem. The only thing I could think of was to check whether the uninstall transaction contains only unused refs and if it does, then allow it.

An alternative would be to create a dedicated transaction function for the purge itself. Then the libflatpak will take care what it'll get rid of (possibly using the code outlined in the description).
